libraries/Microsoft.Bot.Connector/Authentication/PasswordServiceClientCredentialFactory.cs

```diff
@@ -85,30 +108,21 @@ public override Task<ServiceClientCredentials> CreateCredentialsAsync(string app
 
             if (loginEndpoint.StartsWith(AuthenticationConstants.ToChannelFromBotLoginUrlTemplate, StringComparison.OrdinalIgnoreCase))
             {
-                return Task.FromResult<ServiceClientCredentials>(
-                    AppId == null
-                        ?
-                    MicrosoftAppCredentials.Empty
-                        :
-                    new MicrosoftAppCredentials(appId, Password, _httpClient, _logger, oauthScope));
+                return Task.FromResult<ServiceClientCredentials>(string.IsNullOrWhiteSpace(TenantId)
+                    ? new MicrosoftAppCredentials(appId, Password, _httpClient, _logger, oauthScope)
+                    : new MicrosoftAppCredentials(appId, Password, TenantId, _httpClient, _logger, oauthScope));
```

Comment:
can't you just pass in the TenantId on all these?

if its null then that is handled by (if I got this right!?) this line:

https://github.com/microsoft/botbuilder-dotnet/blob/bee20af6ec376de02d46e8d7861edd7344357450/libraries/Microsoft.Bot.Connector/Authentication/AppCredentials.cs#L79

Reply:
Yes, this code can be simplified since the tenant id is being checked for null downstream as you had pointed out above. I have updated the code to pass in tenant id always and ensured the caller is passing in an empty string for multi-tenant and a valid tenant id for single tenant.